sdk/core/System.ClientModel/src/ModelReaderWriter/CollectionReader.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace System.ClientModel.Primitives;
+
+internal abstract class CollectionReader
+{
+    internal static CollectionReader GetCollectionReader(CollectionBuilder builder, ModelReaderWriterOptions options)
+    {
+        if (options.Format != "J" && options.Format != "W")
+        {
+            throw new InvalidOperationException($"Format '{options.Format}' is not supported only 'J' or 'W' format can be read as collections");
+        }
+
+        if (options.Format == "J")
```

Comment:
It would be good to throw a comment in these explaining what "J" and "W" are.  Its not obvious for those unfamiliar with MRW.